src/monodroid/jni/util.h

```diff
@@ -56,6 +58,7 @@ extern "C" {
 #endif // __cplusplus
 
 #define DEFAULT_DIRECTORY_MODE S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH
+#define XA_UNLIKELY(expr) (__builtin_expect ((expr) != 0, 0))
```

Comment:
Should this be wrapped in some form of compiler check?

Reply:
Nope. It works with gcc and clang, that's all we care about. The intrinsic has been around for ages and is supported pretty much everywhere